sources/api/apiserver/src/server/mod.rs

```diff
@@ -354,6 +363,42 @@ async fn get_configuration_files(
     Ok(ConfigurationFilesResponse(resp))
 }
 
+/// Get the update status from 'thar-be-updates'
+async fn get_update_status() -> Result<UpdateStatusResponse> {
+    match thar_be_updates::status::get_update_status() {
+        Ok(update_status) => {
+            let update_status = update_status.simplify_update_info();
+            Ok(UpdateStatusResponse(update_status))
+        }
+        Err(e) => match e {
+            thar_be_updates::error::Error::OpenStatusFile { .. } => {
+                return error::UninitializedUpdateStatus {}.fail()
+            }
+            _ => return error::UpdateError {}.fail(),
+        },
+    }
```

Comment:
This seems like its misusing the accessible functionality in `tbu`. I'm somewhat wary of embedding knowledge and error handling of the tool's internals here. That said, I'm sympathetic to its use and help here.

The only suggestion I can make is to specifically present an "API" to ourselves that namespaces off "safe" functions, types, and helpers to be used in these sort of contexts (eg: `thar_be_updates::api` or `t_b_u::on_disk`).

I'll defer to hear from some of the other rustic folks on this.